DatadogSDKBridge/Classes/Implementation/AnyEncodable.swift

```diff
@@ -6,16 +6,53 @@
 
 import Foundation
 
-internal func castAttributesToSwift(_ attributes: NSDictionary) -> [String: AnyEncodable] {
+internal func castAttributesToSwift(_ attributes: NSDictionary) -> [String: Encodable] {
     return castAttributesToSwift(attributes as? [String: Any] ?? [:])
 }
 
-internal func castAttributesToSwift(_ attributes: [String: Any]) -> [String: AnyEncodable] {
-    return attributes.mapValues { AnyEncodable($0) }
+internal func castAttributesToSwift(_ attributes: [String: Any]) -> [String: Encodable] {
+    var casted: [String: Encodable] = [:]
+
+    attributes.forEach { key, value in
+        if key.hasPrefix("_dd.") {
+            casted[key] = castInternalAttribute(value: value)
+        } else {
+            casted[key] = castUserAttribute(value: value)
+        }
+    }
+
+    return casted
+}
+
+/// Casts `Any` value to `Encodable?` by inspecting and unpacking its value.
+/// This is necessary to handle values passed from Objective-C, which cannot be casted directly (they return `nil` for `as? Encodable`).
+///
+/// **Note**: It only supports selected primitive, non-sequence types.
+private func castInternalAttribute(value: Any) -> Encodable? {
+    switch value {
+    case let string as String: // unpacking `NSTaggedPointerString`
+        return string
+    case let int64 as Int64: // unpacking `__NSCFNumber`
+        return int64
+    case let double as Double: // unpacking `__NSCFNumber` again; trying `Double` after `Int` as it's a wider type
+        return double
+    default:
```

Comment:
💡 I couldn't find a way to handle `Bool` casting in this `switch`. Any `Boolean` value passed from RN would be casted to `Int64`, thus needs to be retrieved with `as? Int64` in `dd-sdk-ios`.

`Int` and `Bool` casts are not exclusive. The value of `0` or `1` is packed as `NSNumber` by Objective-C, so we need to explicitly know which type to unpack. I tried to make it depend on dynamic type check, but [Objective-C's booleans are quite complex](https://nshipster.com/bool/#the-truth-about-nsnumber-and-bool) and nothing seemed reliably for me.

Reply:
This issue is is now fixed 👌 with implementing what was proposed in https://github.com/DataDog/dd-bridge-ios/pull/24#discussion_r660004547